tike/tike.py

```diff
@@ -110,46 +110,50 @@ def admm(
         The energy of the probe
     algorithms : (2, ) string
         The names of the pytchography and tomography reconstruction algorithms.
-    niter : int
+    num_iter : int
         The number of ADMM interations.
+    pkwargs : dict
+        Arguments to pass to the tike.ptycho.reconstruct.
+    tkwargs : dict
+        Arguments to pass to the tike.tomo.reconstruct.
     kwargs :
         Any keyword arguments for the pytchography and tomography
         reconstruction algorithms.
 
     """
+    pkwargs = {
+        'algorithm': 'grad',
+        'num_iter': 1,
+    } if pkwargs is None else pkwargs
+    tkwargs = {
+        'algorithm': 'grad',
+        'num_iter': 1,
+        'reg_par': -1,
+    } if tkwargs is None else tkwargs
     Z, X, Y = obj.shape[0:3]
     T = theta.size
     x = obj
     psi = np.ones([T, Z, Y], dtype=obj.dtype)
     hobj = np.ones_like(psi)
     lamda = np.zeros_like(psi)
-    for i in range(niter):
+    for i in range(num_iter):
         # Ptychography.
         for view in range(len(psi)):
             psi[view] = tike.ptycho.reconstruct(data=data[view],
                                                 probe=probe,
                                                 v=v[view], h=h[view],
                                                 psi=psi[view],
-                                                algorithm='grad',
-                                                niter=1, rho=rho, gamma=gamma,
+                                                rho=rho, gamma=gamma,
                                                 reg=hobj[view],
-                                                lamda=lamda[view], **kwargs)
+                                                lamda=lamda[view], **pkwargs)
         # Tomography.
         phi = -1j / wavenumber(energy) * np.log(psi + lamda / rho) / voxelsize
         x = tike.tomo.reconstruct(obj=x,
                                   theta=theta,
                                   line_integrals=phi,
-                                  algorithm='grad', reg_par=-1,
```

Comment:
You can pass any parameter to the tomography reconstruction algorithm using the `tkwargs` dictionary. Does TomoPy grad have a step length parameter? I don't know. I think that's a question for @math-vrn;  he implemented the gradient descent method for TomoPy.

Reply:
For consistency if ptychography has a step size then tomography should have it too (or vice versa). Generally speaking most iterative algorithms have a ```step_length```, either manually selected or automatically adjusted at each iteration. 

In TomoPy we provide it through ```reg_par``` for passing this information. if it is -1 it is automatically adjusted, but you can provide a fix number as well. But, this is confusing and needs to be changed, because step length is not a regularization parameter. So, I wouldn't make assumptions on the parameter list by checking the practices in child packages for ptychography and tomography.

Reply:
:confused: I didn't design the API for tomopy.grad, but I call it from tike, so I picked default values that makes sense. If you want the tomopy API to change, then you should open an Issue over there. There's nothing I can do about the TomoPy API in this pull request.